rustls/src/msgs/deframer.rs

```diff
@@ -46,25 +46,12 @@ impl MessageDeframer {
     /// Returns an `Error` if the deframer failed to parse some message contents,
     /// `Ok(None)` if no full message is buffered, and `Ok(Some(_))` if a valid message was found.
     pub fn pop(&mut self) -> Result<Option<OpaqueMessage>, Error> {
-        match self.desynced {
-            false => Ok(self.frames.pop_front()),
-            true => Err(Error::CorruptMessage),
+        if self.desynced {
+            return Err(Error::CorruptMessage);
+        } else if let Some(msg) = self.frames.pop_front() {
```

Comment:
Added a fast path here where we don't try to parse again if there are still parsed messages in the queue.